jetty-deploy/src/main/java/org/eclipse/jetty/deploy/bindings/StandardStarter.java

```diff
@@ -35,12 +36,17 @@
     @Override
     public void processBinding(Node node, App app) throws Exception
     {
+        ContextHandlerCollection contexts = app.getDeploymentManager().getContexts();
+
         ContextHandler handler = app.getContextHandler();
-        
-        // start the handler
-        handler.start();
-        
-        // After starting let the context manage state
-        app.getDeploymentManager().getContexts().manage(handler);
+
+        if (contexts.isRunning() && !handler.isRunning())
```

Comment:
Change to 
```
if (contexts.isStarted() && handler.isStopped())
```
so failed handlers are not started.

Reply:
Current test cases for this PR show that failed handlers are not restarted again.

Reply:
However, I believe that the `contexts.isStarted() && handler.isStopped()` form is clearer and more correct. It avoids untested corner cases of starting and stopping states.

Reply:
`context.isRunning()` seems more appropriate as its for both STARTED and STARTING.

Reply:
Tests with `handler.isStopped()` are causing a deployment loop if `WebAppContext.isThrowUnavailableOnStartupException == false` now.
Troubleshooting ...

Reply:
Seems that the handler with `isThrowUnavailableOnStartupException == false` goes in state ...

1. STARTING
2. (error occurs, but is not reported due to `isThrowUnavailableOnStartupException == false`)
3. STOPPED

If the same handler with setting `isThrowUnavailableOnStartupException == true` then it seems to go into ...

1. STARTING
2. (error occurs, flows out and is caught by state handling)
3. FAILED